### PR TITLE
Improve the CMS

### PR DESCRIPTION
**Pull request by tgalopin**

### CODE REVIEW

File: src/AppBundle/Entity/ArticleCategory.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(length=50)
+     *
+     * @Assert\NotBlank
```
Comment (HeahDude) [inline src/AppBundle/Entity/ArticleCategory.php]: Missing `@Assert\Length(max=50)`?

File: src/AppBundle/Entity/EntityContentTrait.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+use Symfony\Component\Validator\Constraints as Assert;
+
+trait EntityContentTrait
+{
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=100)
+     *
+     * @Assert\NotBlank
```
Comment (HeahDude) [inline src/AppBundle/Entity/EntityContentTrait.php]: Same as above and below, perhaps?

File: src/AppBundle/Entity/EntityContentTrait.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+     */
+    public function getTitle()
+    {
+        return $this->title;
+    }
+
+    /**
+     * @param null|string $title
+     *
+     * @return EntityContentTrait
```
Comment (HeahDude) [inline src/AppBundle/Entity/EntityContentTrait.php]: `$this` same below, IDEs will resolve it dynamically then.

File: src/AppBundle/Entity/LiveLink.php [hunk 4]
```diff
@@ -72,7 +72,7 @@ public function getPosition()
         return $this->position;
     }
 
-    public function setPosition($position)
+    public function setPosition($position): LiveLink
```
Comment (HeahDude) [inline src/AppBundle/Entity/LiveLink.php]: Can be `:self`.

File: src/AppBundle/Entity/Proposal.php [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    {
+        return $this->title;
+    }
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    public function getPosition()
```
Comment (HeahDude) [inline src/AppBundle/Entity/Proposal.php]: `: ?int`?

File: src/AppBundle/Entity/Proposal.php [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+     * @var int
+     *
+     * @ORM\Column(type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var int
```
Comment (HeahDude) [inline src/AppBundle/Entity/Proposal.php]: `|null`

File: src/AppBundle/Entity/Proposal.php [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    {
+        $this->published = $published;
+
+        return $this;
+    }
+
+    /**
+     * @return Media|null
+     */
+    public function getMedia()
```
Comment (HeahDude) [inline src/AppBundle/Entity/Proposal.php]: `: ?Media`

File: src/AppBundle/Entity/Proposal.php [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+    {
+        return $this->media;
+    }
+
+    /**
+     * @param Media|null $media
+     *
+     * @return Proposal
+     */
+    public function setMedia(Media $media = null): Proposal
```
Comment (HeahDude) [inline src/AppBundle/Entity/Proposal.php]:
```php
public function setMedia(?Media $media): self
```

File: src/AppBundle/Entity/Proposal.php~ [hunk 9]
```diff
@@ -0,0 +1,132 @@
+<?php
```
Comment (HeahDude) [inline src/AppBundle/Entity/Proposal.php~]: Does it really have to be committed?

File: src/AppBundle/Entity/ProposalTheme.php [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table(name="proposals_themes")
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\ProposalThemeRepository")
+ */
+class ProposalTheme
+{
+    /**
+     * @var int
```
Comment (HeahDude) [inline src/AppBundle/Entity/ProposalTheme.php]: `|null`

File: src/AppBundle/Entity/ProposalTheme.php [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+        $this->name = $name;
+        $this->color = $color;
+    }
+
+    public function __toString()
+    {
+        return $this->name;
+    }
+
+    public function getId(): int
```
Comment (HeahDude) [inline src/AppBundle/Entity/ProposalTheme.php]: `: ?int`

File: src/AppBundle/Repository/ProposalRepository.php [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+            ->leftJoin('p.themes', 't')
+            ->orderBy('p.position', 'ASC')
+            ->getQuery()
+            ->getResult();
+    }
+
+    /**
+     * @param string $slug
+     *
+     * @return Proposal
```
Comment (HeahDude) [inline src/AppBundle/Repository/ProposalRepository.php]: `|null`

**Comment by tgalopin** [comment]
Thanks, I'll fix them asap.

**Pull request merged**